internal/resolution/variablesources/bundle.go

```diff
@@ -2,62 +2,36 @@ package variablesources
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
-	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
-var _ input.VariableSource = &BundlesAndDepsVariableSource{}
-
-type BundlesAndDepsVariableSource struct {
-	catalogClient   BundleProvider
-	variableSources []input.VariableSource
-}
-
-func NewBundlesAndDepsVariableSource(catalogClient BundleProvider, inputVariableSources ...input.VariableSource) *BundlesAndDepsVariableSource {
-	return &BundlesAndDepsVariableSource{
-		catalogClient:   catalogClient,
-		variableSources: inputVariableSources,
-	}
-}
-
-func (b *BundlesAndDepsVariableSource) GetVariables(ctx context.Context) ([]deppy.Variable, error) {
-	var variables []deppy.Variable
-
-	// extract required package variables
-	for _, variableSource := range b.variableSources {
-		inputVariables, err := variableSource.GetVariables(ctx)
-		if err != nil {
-			return nil, err
-		}
-		variables = append(variables, inputVariables...)
-	}
-
-	// create bundle queue for dependency resolution
+func MakeBundleVariables(
+	allBundles []*catalogmetadata.Bundle,
+	requiredPackages []*olmvariables.RequiredPackageVariable,
+	installedPackages []*olmvariables.InstalledPackageVariable,
+) ([]*olmvariables.BundleVariable, error) {
 	var bundleQueue []*catalogmetadata.Bundle
```

Comment:
Allocate with an initial capacity:
```suggestion
	bundleQueue := make([]*catalogmetadata.Bundle, 0, len(requiredPackages) + len(installedPackages))
```

Reply:
I think here we should not do that becuase `bundleQueue` is for bundles on which `requiredPackages` and `installedPackages` variables depend and for dependencies of these bundles.

`len(requiredPackages) + len(installedPackages)` will not correspond to the amount of data we handle and the initial capacity will not be meaningful.

Reply:
Another reason is that - I think it is a bit confusing. When I see a `make` like this - it looks like we allocating a new bigger slice to accomadate both `requiredPackages` and `installedPackages`. But it is not the case.

Reply:
I didn't look closely enough to see if there's a minimum bound on the high-water mark of the `bundleQueue`. My general point is: "let's always pre-allocate if the minimum high-water mark is greater than 0".

If, in this case, that minimum high-water mark is 0, then I withdraw my suggestion.

Reply:
Even if the slice can end up being bigger than the minimum bound, it still saves on allocations and copies for all of the intermediate allocations/copies that we would have made otherwise.

Reply:
On this function's level - it is possible that it will be 0, but looking at the bigger picture  - I think will never reach this function without bundles for `requiredPackages` and `installedPackages`. Each item in these slices should have >= 1 bundle.

Updated.